### PR TITLE
Lazy transaction record docs

### PR DESCRIPTION
**Pull request by sploiselle**
@nvanbenschoten Can I get you to take a look at this for a tech edit? Still need to polish the language a bit, but want to make sure I have the technical details lined up correctly. Oh––and one open question: does the `DistSender` or the `TxnCoordSender` actually write the transaction record?

Closes #4397

### CODE REVIEW

File: v2.2/architecture/life-of-a-distributed-transaction.md [hunk 1]
```diff
@@ -142,6 +134,11 @@ If an operation encounters a write intent for a key, it attempts to "resolve" th
   If the push succeeds, the operation continues.
 
   If this push fails (which is the majority of the time), this transaction goes into the [`TxnWaitQueue`](https://www.cockroachlabs.com/docs/stable/architecture/transaction-layer.html#txnwaitqueue) on this node. The incoming transaction can only continue once the blocking transaction completes (i.e., commits or aborts).
+- _Missing_, the resolver checks the write intent's timestamp. If it was created within the last two seconds, it exhibits the `PENDING` behavior, with the addition of tracking the push in the range's timestamp cache, which will inform the transaction that its timestamp was pushed once the transaction record gets created.
```
Comment (nvanbenschoten) [inline v2.2/architecture/life-of-a-distributed-transaction.md]: s/checks/consults/

s/it exhibits/it treats the transaction record as exhibiting/

File: v2.2/architecture/life-of-a-distributed-transaction.md [hunk 2]
```diff
@@ -179,12 +176,14 @@ Note that this write operation creates a write intent; these writes will not be
...
-	- If the transaction's timestamp was moved, the transaction performs a [read refresh](transaction-layer.html#read-refreshing) to see if any values it needed have been changed. If the read refresh is successful, the transaction can commit at the pushed timestamp. If the read refresh fails, the transaction must be restarted.
+1. Once all operations have successfully completed (i.e. reads have returned values and write intents have been committed), the `DistSender` tries to record the transaction's success in the transaction record (which provides a durable mechanism of tracking the transaction's state), which can cause a few situations to arise:
+    - It check's the timestamp cache of the range where the first write occurred to see if its timestamp got pushed forward. If it did, the transaction performs a [read refresh](transaction-layer.html#read-refreshing) to see if any values it needed have been changed. If the read refresh is successful, the transaction can commit at the pushed timestamp. If the read refresh fails, the transaction must be restarted.
```
Comment (nvanbenschoten) [inline v2.2/architecture/life-of-a-distributed-transaction.md]: s/check's/checks/

File: v2.2/architecture/transaction-layer.md [hunk 3]
```diff
@@ -102,21 +102,30 @@ After setting up this bookkeeping, the request is passed to the `DistSender` in
 
 ### Transaction records
 
-When a transaction starts, `TxnCoordSender` writes a transaction record to the range containing the first key modified in the transaction. As mentioned above, the transaction record provides the system with a source of truth about the status of a transaction.
+To track the status of a transaction's execution, we write a value called a transaction record to our key-value store. This lets any transaction check the status of any other transaction, which is important for handling concurrency.
 
-The transaction record expresses one of the following dispositions of a transaction:
+When a `TxnCoordSender` begins transaction, it determines where to write the transaction record (this location will always be the same range as the first key in the transaction), and then includes a reference to the transaction record in all of its write intents.
```
Comment (nvanbenschoten) [inline v2.2/architecture/transaction-layer.md]: begins **a/its** transaction

File: v2.2/architecture/transaction-layer.md [hunk 4]
```diff
@@ -102,21 +102,30 @@ After setting up this bookkeeping, the request is passed to the `DistSender` in
...
+- The write operation commits
+- The `TxnCoordSender` heartbeats the transaction
+- An operation forces the transaction to abort
+
+Given this mechanism, the transaction record uses the following states:
+
+- `PENDING`: Indicates that the write intent's transaction is still in progress.
+- `COMMITTED`: Once a transaction has completed, this status indicates that write intents can be treated as committed values.
+- `ABORTED`: Indicates that the transaction was aborted and its values should be discarded.
+- _Record does not exist_: Because records might not have been written yet, a missing record require the contending transaction to look at the write intent's timestamp. If it was created within the last two seconds, the transaction is treated as if it's `PENDING`, otherwise the transaction is considered `ABORTED`.
```
Comment (nvanbenschoten) [inline v2.2/architecture/transaction-layer.md]: We might want to specify what this write intent is. Probably a condensed version of "the write intent that the contending transaction observed which drove it to look for the transaction record."

Comment (sploiselle) [reply v2.2/architecture/transaction-layer.md]: Tried to fix this up with a better explanation and better pronouns. Can I get you to take a quick look?

**Comment by sploiselle** [comment]
@rmloveland Ready for copy edits

**Comment by jseldess** [comment]
@rmloveland, can you review this for Sean so we can get it merged?

**Comment by rmloveland** [comment]
Right away! Sorry, must've dropped it somehow

File: v2.2/architecture/life-of-a-distributed-transaction.md [hunk 5]
```diff
@@ -179,12 +176,14 @@ Note that this write operation creates a write intent; these writes will not be
...
+    - It checks the timestamp cache of the range where the first write occurred to see if its timestamp got pushed forward. If it did, the transaction performs a [read refresh](transaction-layer.html#read-refreshing) to see if any values it needed have been changed. If the read refresh is successful, the transaction can commit at the pushed timestamp. If the read refresh fails, the transaction must be restarted.
 	- If the transaction is in an `ABORTED` state, the `DistSender` sends a response indicating as much, which ends up back at the SQL interface.
-	- If the transaction is still `PENDING`, it's moved to `COMMITTED`. Note that this modification of the transaction record is handled as an update to the key-value pair, meaning it must go through Raft like all other write operations. Once the transaction record is moved to `COMMITTED`, the transaction is considered committed.
+
+	Upon passing these checks the transaction record is either written for the first time with the `COMMITTED` state, or if it was in a `PENDING` state, it is moved to `COMMITTED`. Only at this point is the transaction is considered commited.
```
Comment (rmloveland) [inline v2.2/architecture/life-of-a-distributed-transaction.md]: Typos: extra "is", two t's in "committed".  Suggest:

"Only at this point is the transaction considered committed"

**Comment by jseldess** [comment]
@sploiselle, any chance you can polish this one off this week?

**Pull request merged**